### PR TITLE
Add pull-kubernetes-dependencies-canary job

### PR DESCRIPTION
**Pull request by cblecker**
/cc @liggitt @BenTheElder

### CODE REVIEW

File: config/jobs/kubernetes/sig-testing/dependencies.yaml [hunk 1]
```diff
@@ -30,6 +30,36 @@ presubmits:
...
+        securityContext:
+          privileged: true
+        env:
+        # Space separated list of the checks to run
+        - name: WHAT
+          value: "vendor vendor-licenses"
+        - name: GOPROXY
+          value: https://proxy.golang.org
+        - name: GOSUMDB
+          value: sum.golang.org
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-testing/dependencies.yaml]: I think this won't take affect with our go version, but can't hurt to have.
https://groups.google.com/forum/#!msg/golang-dev/go6_lZqew6g/rNg3KacMDQAJ

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/dependencies.yaml]: (GOSUMDB that is)

**Comment by liggitt** [comment]
the GOPROXY envvar lgtm, not sure what the rest does

**Comment by cblecker** [comment]
@liggitt It's basically a copy of the main job, but without reporting and won't be merge blocking. That way we can test out and see if this helps with pull times.

The other thing I want to do is create a cross build compile job canary, and use the cache and disable vendor.

**Comment by BenTheElder** [comment]
FYI katharine is now using the proxy for some slack-infra jobs

**Comment by liggitt** [comment]
> The other thing I want to do is create a cross build compile job canary, and use the cache and disable vendor.

we'll need to actually switch our build scripts to use modules first (which we should definitely do), with GOFLAGS defaulting to `-mod=vendor`, then the cross build job can override GOFLAGS to disable vendor and it will actually take effect

basically, once this is removed (or pinned to `GO111MODULE=on`), we can run that job:
https://github.com/kubernetes/kubernetes/blob/9466118e8fa6deaa719b3ec8a03bb6eb846a76fc/hack/lib/init.sh#L25-L29

**Comment by cblecker** [comment]
@liggitt gotcha. we need to move off of GOPATH stuff entirely from our build scripts before the module stuff will work?

**Comment by liggitt** [comment]
> @liggitt gotcha. we need to move off of GOPATH stuff entirely from our build scripts before the module stuff will work?

yes

**Comment by cblecker** [comment]
/assign @spiffxp 
for approval

**Comment by cblecker** [comment]
/assign @fejta 
pinging more peopleeeee

**Pull request merged**